# HG changeset patch
# User Yuya Nishihara <yuya@tcha.org>
# Date 1319557903 -32400
# Node ID f3817a7abb17cb14bdbb428b63345fb2d17f0853
# Parent  08559b0a658f824b99294ac12cd2e472f4cd5129
hgversion: carry minor digit to major like 1.9 to 2.0

diff --git a/tortoisehg/util/hgversion.py b/tortoisehg/util/hgversion.py
--- a/tortoisehg/util/hgversion.py
+++ b/tortoisehg/util/hgversion.py
@@ -26,8 +26,7 @@
     vers = re.split(r'\.|-', v)[:2]
     if vers == reqver or len(vers) < 2:
         return
-    nextver = list(reqver)
-    nextver[1] = str(int(reqver[1])+1)
+    nextver = map(str, divmod(int(reqver[0]) * 10 + int(reqver[1]) + 1, 10))
     if vers == nextver:
         return
     return (('This version of TortoiseHg requires Mercurial '
